source/FreeRTOS_ND.c

```diff
@@ -1314,4 +1334,87 @@
+        /* More details at: https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/MISRA.md#rule-113 */
+        /* coverity[misra_c_2012_rule_11_3_violation] */
+        IPPacket_IPv6_t * pxIPPacket = ( ( IPPacket_IPv6_t * ) pxNetworkBuffer->pucEthernetBuffer );
+        IPHeader_IPv6_t * pxIPHeader = &( pxIPPacket->xIPHeader );
+        IPv6_Address_t * pxIPAddress = &( pxIPHeader->xSourceAddress );
+        uint8_t ucNextHeader = pxIPHeader->ucNextHeader;
+
+        configASSERT( pxIPPacket->xEthernetHeader.usFrameType == ipIPv6_FRAME_TYPE );
+
+        if( ( ucNextHeader == ipPROTOCOL_TCP ) ||
+            ( ucNextHeader == ipPROTOCOL_UDP ) )
+        {
+            IPv6_Type_t eType = xIPv6_GetIPType( ( const IPv6_Address_t * ) pxIPAddress );
+            FreeRTOS_debug_printf( ( "xCheckRequiresNDResolution: %pip type %s\n",
+                                     ( void * ) pxIPAddress->ucBytes,
+                                     ( eType == eIPv6_Global ) ? "Global" :
+                                     ( eType == eIPv6_LinkLocal ) ? "LinkLocal" :
+                                     ( eType == eIPv6_Loopback ) ? "Loopback" :
+                                     "other" ) );
+
+            if( eType == eIPv6_LinkLocal )
+            {
+                MACAddress_t xMACAddress;
+                NetworkEndPoint_t * pxEndPoint;
+                eResolutionLookupResult_t eResult;
+                char pcName[ 80 ];
+
+                ( void ) memset( &( pcName ), 0, sizeof( pcName ) );
+                eResult = eNDGetCacheEntry( pxIPAddress, &xMACAddress, &pxEndPoint );
+                FreeRTOS_printf( ( "xCheckRequiresNDResolution: eResult %s with EP %s\n", ( eResult == eResolutionCacheMiss ) ? "Miss" : ( eResult == eResolutionCacheHit ) ? "Hit" : "Error", pcEndpointName( pxEndPoint, pcName, sizeof pcName ) ) );
+
+                if( eResult == eResolutionCacheMiss )
+                {
+                    NetworkBufferDescriptor_t * pxTempBuffer;
+                    size_t uxNeededSize;
+
+                    uxNeededSize = sizeof( ICMPPacket_IPv6_t );
+                    pxTempBuffer = pxGetNetworkBufferWithDescriptor( BUFFER_FROM_WHERE_CALL( 199 ) uxNeededSize, 0U );
+
+                    if( pxTempBuffer != NULL )
+                    {
+                        pxTempBuffer->pxEndPoint = pxNetworkBuffer->pxEndPoint;
+                        pxTempBuffer->pxInterface = pxNetworkBuffer->pxInterface;
+                        vNDSendNeighbourSolicitation( pxTempBuffer, pxIPAddress );
+                    }
+
+                    xNeedsNDResolution = pdTRUE;
+                }
+            }
+        }
+
+        return xNeedsNDResolution;
+    }
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Send an unsolicited ND packet to allow this node to announce the IP-MAC
+ *        mapping to the entire network.
+ */
+    void vNDSendUnsolicited( void )
```

Comment:
I dont see the function `vNDSendUnsolicited` used anywhere. Is it supposed to be used by the application writer?

Reply:
I will remove it. I was going to implement it as the ND variant of a gratuitous arp but decided against it